rgdps/usecases/friend_requests.py

```diff
@@ -0,0 +1,233 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import NamedTuple
+
+from rgdps import repositories
+from rgdps.common.context import Context
+from rgdps.constants.errors import ServiceError
+from rgdps.constants.users import UserRelationshipType
+from rgdps.models.friend_request import FriendRequest
+from rgdps.models.user import User
+
+
+class FriendRequestResponse(NamedTuple):
+    request: FriendRequest
+    user: User
+
+
+class PaginatedFriendRequestResponse(NamedTuple):
+    requests: list[FriendRequestResponse]
+    total: int
+
+
+async def get_user(
+    ctx: Context,
+    user_id: int,
+    page: int = 0,
+    page_size: int = 10,
+    is_sender_user_id: bool = False,
+) -> PaginatedFriendRequestResponse | ServiceError:
+
+    requests = await repositories.friend_requests.from_user_id_paginated(
+        ctx,
+        user_id,
+        page,
+        page_size,
+        is_sender_user_id=is_sender_user_id,
+        include_deleted=False,
+    )
+
+    friend_request_responses = []
+    for request in requests:
+        # Swap the sender and recipient according to the from_sender_id flag
+        target_user_id = (
+            request.recipient_user_id if is_sender_user_id else request.sender_user_id
+        )
+        user = await repositories.user.from_id(ctx, target_user_id)
+
+        if user is None:
+            continue
+
+        friend_request_responses.append(
+            FriendRequestResponse(request, user),
+        )
+
+    friend_request_count = (
+        await repositories.friend_requests.get_user_friend_request_count(
+            ctx,
+            user_id,
+            is_sender_user_id=is_sender_user_id,
+        )
+    )
```

Comment:
Is there a need for these extra brackets?

Reply:
they are caused by black formatter, we agreed to leave it as it is in dms.